src/components/data-management/DownloadData.jsx

```diff
@@ -0,0 +1,112 @@
+import React from 'react';
+import _ from 'lodash';
+import { Menu, Tooltip } from 'antd';
+import { useSelector } from 'react-redux';
+import PropTypes from 'prop-types';
+import { saveAs } from 'file-saver';
+import pipelineStatus from '../../utils/pipelineStatusValues';
+import downloadData from '../../utils/data-management/downloadExperimentData';
+import downloadTypes from '../../utils/data-management/downloadTypes';
+import { exportQCParameters, filterQCParameters } from '../../utils/exportQCParameters';
+
+const DownloadData = (props) => {
+  const {
+    activeProjectUuid,
+  } = props;
+  const activeProject = useSelector((state) => state.projects[activeProjectUuid]);
+  const experimentSettings = useSelector((state) => state.experimentSettings);
+  const experiments = useSelector((state) => state.experiments);
+  const samples = useSelector((state) => state.samples);
+
+  const pipelineHasRun = (experimentId) => (
+    experiments[experimentId]?.meta?.backendStatus?.pipeline?.status === pipelineStatus.SUCCEEDED
+  );
+  const gem2sHasRun = (experimentId) => (
+    experiments[experimentId]?.meta?.backendStatus?.gem2s?.status === pipelineStatus.SUCCEEDED
+  );
```

Comment:
since every time you also have a check for `activeProject?.experiments?.length > 0` when you use `pipelineHasRun` and `gem2sHasRun`, I would recommend that you move this check: `activeProject?.experiments?.length > 0` to be part of these statements

Reply:
Why do you actually have to check that too? Wouldn't it be sufficient if `experimentId` is in `experiments`?

Reply:
We need these checks to determine whether we can download :
raw seurat object - needs gem2s to have finished
processed seurat object - needs pipeline to have finished